File: python/yax/test/saxtest.py
#This software is released under the terms of the Apache License version 2.
#For details of the license, see http://www.apache.org/licenses/LICENSE-2.0.

import os
import sys
import getopt
import codecs
import xml.sax
import StringIO
sys.path.append(os.path.join(os.getcwd(),"yax"))

import saxeventhandler
from saxevent import SaxEvent
from saxeventhandler import SaxEventHandler
import saxeventtype
import util
import testoptions

EXPECTED_VERSION = "2.00"

# Simple subclass of SaxEventHandler
class SaxTestHandler(SaxEventHandler) :

  def __init__(self, document) :
    SaxEventHandler.__init__(self,document)

  def orderedAttributes(self, element) : return None

  # Overide super class methods
  def yyevent(self, token):
    trace = util.saxtrace(token,self.flags);
    sys.stdout.write("saxtest: {0}\n".format(trace))
    sys.stdout.flush();
  # end yyevent
#end class SaxTestHandler

def main():

  if(EXPECTED_VERSION != saxeventhandler.getVersion()):
    sys.stderr.write("Version mismatch: {0} :: {1}\n",
                     EXPECTED_VERSION, saxeventhandler.getVersion())
    sys.exit(1)

  testoptions.getOptions(sys.argv);

  # Create handler
  handler = SaxTestHandler(testoptions.document)
  handler.setFlags(testoptions.flags)
  if(testoptions.saxtrace):
    handler.setTrace(True)

  # Parse the document
  handler.parse()

  sys.stdout.write("No error\n");
  sys.exit(0);
# end main()

if __name__ == '__main__':
  main()

File: java/src/yax/test/SaxTest.java
/**
 This software is released under the terms of the Apache License version 2.
 For details of the license, see http://www.apache.org/licenses/LICENSE-2.0.
 */

package yax.test;

import yax.*;

import org.w3c.dom.*;
import org.xml.sax.*;

import java.io.*;

public class SaxTest
{

    static final String EXPECTED_VERSION = "2.00";

    // Simple subclass of SaxEventHandler
    static public class SaxTestHandler extends SaxEventHandler
    {
        public SaxTestHandler(String document) throws SAXException
        {
            super(document);
        }

        // Define the abstract methods
        public String[] orderedAttributes(String element)
        {
            return null;
        }

        public void yyevent(SaxEvent token)
            throws SAXException
        {
            String trace = null;
            trace = Util.trace(token, getFlags());
            System.out.printf("saxtest: %s\n", trace);
            System.out.flush();
        }
    }

    static public void
    main(String[] argv)
    {
        try {
            SaxTestHandler handler;
            SaxEventType tokentype = null;

	    if(!EXPECTED_VERSION.equals(SaxEventHandler.getVersion())) {
		System.err.printf("Version mismatch: %s :: %s\n",
			EXPECTED_VERSION, SaxEventHandler.getVersion());
		System.exit(1);
	    }
            TestOptions.getoptions(argv);
            handler = new SaxTestHandler(TestOptions.document);
            handler.setFlags(TestOptions.flags);
            if(TestOptions.saxtrace)
                handler.setTrace(true);
            handler.parse();
        } catch (Exception e) {
            e.printStackTrace();
            System.exit(1);
        }
        System.out.println("No error");
        System.exit(0);
    }

} //SaxTest
